backend/controller/timeline/events_ingress.go

```diff
@@ -3,31 +3,141 @@ package timeline
 import (
 	"context"
 	"encoding/json"
+	"io"
+	"net/http"
+	"time"
 
 	"github.com/alecthomas/types/optional"
 
+	ftlencryption "github.com/TBD54566975/ftl/backend/controller/encryption/api"
 	"github.com/TBD54566975/ftl/backend/controller/timeline/internal/sql"
 	"github.com/TBD54566975/ftl/backend/libdal"
+	"github.com/TBD54566975/ftl/backend/schema"
+	"github.com/TBD54566975/ftl/internal/log"
 	"github.com/TBD54566975/ftl/internal/model"
 )
 
-type IngresEvent struct {
+type IngressEvent struct {
+	ID            int64
 	DeploymentKey model.DeploymentKey
-	RequestKey    model.RequestKey
+	RequestKey    optional.Option[model.RequestKey]
+	Verb          schema.Ref
+	Method        string
+	Path          string
+
+	StatusCode     int
+	Time           time.Time
+	Duration       time.Duration
+	Request        json.RawMessage
+	RequestHeader  json.RawMessage
+	Response       json.RawMessage
+	ResponseHeader json.RawMessage
+	Error          optional.Option[string]
 }
 
-// The internal JSON payload of an ingress event.
+func (e *IngressEvent) GetID() int64 { return e.ID }
+func (e *IngressEvent) event()       {}
+
 type eventIngressJSON struct {
-	DurationMS int64                   `json:"duration_ms"`
-	Request    json.RawMessage         `json:"request"`
-	Response   json.RawMessage         `json:"response"`
-	Error      optional.Option[string] `json:"error,omitempty"`
-	Stack      optional.Option[string] `json:"stack,omitempty"`
+	DurationMS     int64                   `json:"duration_ms"`
+	Method         string                  `json:"method"`
+	Path           string                  `json:"path"`
+	StatusCode     int                     `json:"status_code"`
+	Request        json.RawMessage         `json:"request"`
+	RequestHeader  json.RawMessage         `json:"req_header"`
+	Response       json.RawMessage         `json:"response"`
+	ResponseHeader json.RawMessage         `json:"res_header"`
```

Comment:
ultra nitty nit: I don't think I've seen `response` abbreviated to `res` - only to `resp`. It is annoying that `resp` is 4 long while `req` is 3, so they don't match nicely then, but if I saw `res_header` with no context, I probably wouldn't guess what it is. What do you think? No strong feelings on my part